scripts/autotools/libmongoc/CheckICU.m4

```diff
@@ -0,0 +1,40 @@
+PHP_ARG_WITH([mongodb-icu],
+             [whether to enable ICU for SASLPrep with SCRAM-SHA-256 authentication],
+             [AS_HELP_STRING([--with-mongodb-icu=@<:@auto/yes/no@:>@],
+                             [MongoDB: Enable ICU for SASLPrep with SCRAM-SHA-256 authentication [default=auto]])],
+             [auto],
+             [no])
+
+AS_IF([test "$PHP_MONGODB_ICU" = "auto" -o "$PHP_MONGODB_ICU" = "yes"],[
+  found_icu=no
+
+  PKG_CHECK_MODULES([PHP_MONGODB_ICU],[icu-uc],[
+    PHP_MONGODB_BUNDLED_CFLAGS="$PHP_MONGODB_BUNDLED_CFLAGS $PHP_MONGODB_ICU_CFLAGS"
+    PHP_EVAL_LIBLINE([$PHP_MONGODB_ICU_LIBS],[MONGODB_SHARED_LIBADD])
+    PHP_MONGODB_ICU="yes"
+	found_icu=yes
+  ],[
+	found_icu=no
+  ])
+
+  if test "$PHP_MONGODB_ICU" = "auto" -a "$found_icu" = "no"; then
+    AC_MSG_WARN([ICU libraries and development headers could not be found])
```

Comment:
Is there a particular reason we decided to do this? AFAIK other dependencies don't warn if "auto" fails to find anything.

There should be a pair of checking/result macros down below to report what we found, such as:

```
AC_MSG_CHECKING([which SASL library to use])
AC_MSG_RESULT([$PHP_MONGODB_SASL])
```

Reply:
We do the warning for Cyrus SASL too: https://github.com/mongodb/mongo-php-driver/blob/master/scripts/autotools/libmongoc/CheckSasl.m4#L41-L43 — I'll remove that though.

I didn't see the need for doing a "which SASL library to use" bit, as there is no choice being made. The output already has:

```
checking whether to enable ICU for SASLPrep with SCRAM-SHA-256 authentication... auto
checking for PHP_MONGODB_ICU... yes
```

Reply:
> We do the warning for Cyrus SASL too: https://github.com/mongodb/mongo-php-driver/blob/master/scripts/autotools/libmongoc/CheckSasl.m4#L41-L43 — I'll remove that though.

The bit in CheckSasl.m4 is not a warning. We purposefully raise an error if "cyrus" was explicitly selected but not found. This is comparable to the `AC_MSG_ERROR()` you're doing a few lines below here.

> The output already has:

`checking for PHP_MONGODB_ICU... yes` is emitted from `PKG_CHECK_MODULES`. I suppose the extra `AC_MSG_CHECKING()` would be needlessly verbose here because ICU doesn't have multiple options (unlike compression, SASL, SSL). Therefore, `checking for PHP_MONGODB_ICU` tells us all we need to know.

Reply:
Ah, I didn't realize "I'll remove that though." was referring to the `AC_MSG_WARN()` for ICU and thought you were going to remove the linked line in CheckSasl.m4. Based on the current diff, all looks good.